Server/src/bgp/linkstate/MPLinkStateAttr.cpp

```diff
@@ -145,20 +166,7 @@ namespace bgp_msg {
                              peer_addr.c_str(), len);
                 }
 
-                std::string flags = "";
-
-                if (*data & NODE_FLAG_MASK_OVERLOAD)
-                    flags += "O";
-                if (*data & NODE_FLAG_MASK_ATTACH)
-                    flags += "T";
-                if (*data & NODE_FLAG_MASK_EXTERNAL)
-                    flags += "E";
-                if (*data & NODE_FLAG_MASK_ABR)
-                    flags += "B";
-                if (*data & NODE_FLAG_MASK_ROUTER)
-                    flags += "R";
-                if (*data & NODE_FLAG_MASK_V6)
-                    flags += "V";
+                std::string flags = this->parse_flags_to_string(*data, std::array<std::string, 8>{"O", "T", "E", "B", "R", "V"});
```

Comment:
Can you define each of these flags in header file using the descriptive label "NODE_FLAG_..." and then static init them there?   You can then call the parse_flags_to_string() method using the defined flag array.